src/tests/src/solver/simulation/remix-storage/tests-multi-storage-remix.cpp

```diff
@@ -0,0 +1,266 @@
+        levels.assign(size, 0.);
+        pmax.assign(size, std::numeric_limits<double>::max());
+        inflows.assign(size, 0.);
+        ovf.assign(size, 0.);
+        lowRuleCurve.assign(size, 0.);
+        upRuleCurve.assign(size, capacity);
+    }
+
+    std::shared_ptr<IStorageForRemix> createSTS(std::vector<double>& unsupE)
+    {
+        return makeSTSforRemix(withdrawal,
+                               unsupE,
+                               levels,
+                               pmax,
+                               inflows,
+                               injection,
+                               lowRuleCurve,
+                               upRuleCurve,
+                               initLevel,
+                               withdrawalEff,
+                               injectionEff);
+    }
+
+    std::vector<double> withdrawal, injection, levels, pmax, inflows, ovf;
+    std::vector<double> lowRuleCurve, upRuleCurve;
+
+    double initLevel = 0.;
+    double capacity = std::numeric_limits<double>::max();
+    const double withdrawalEff = 1.;
+    const double injectionEff = 1.;
+};
+
+template<unsigned int size>
+struct InputFixture
+{
+    InputFixture()
+    {
+        Load.assign(size, 0.);
+        TotaGenWithoutStorage.assign(size, 0.);
+        UnsupE.assign(size, 0.);
+        Spillage.assign(size, 0.);
+        DTG_MRG.assign(size, 0.);
+    }
+
+    void runRemixStorageAlgorithm()
+    {
+        Load = TotaGenWithoutStorage + UnsupE + sts_1.withdrawal + sts_2.withdrawal;
+
+        storagesForRemix.clear();
+        storagesForRemix.push_back(sts_1.createSTS(UnsupE));
+        storagesForRemix.push_back(sts_2.createSTS(UnsupE));
+        shavePeaksByRemixingStorageGen(Load, UnsupE, Spillage, DTG_MRG, storagesForRemix);
+    }
+
+    STS_setup<size> sts_1;
+    STS_setup<size> sts_2;
+
+    std::vector<double> Load, TotaGenWithoutStorage, UnsupE, Spillage, DTG_MRG;
+
+    ListStorageForRemix storagesForRemix;
+};
+
+BOOST_FIXTURE_TEST_CASE(default_input_in_STS_setup_does_not_raise_exception, STS_setup<5>)
+{
+    std::vector<double> unsupE(5, 0.);
+    BOOST_CHECK_NO_THROW(createSTS(unsupE));
+}
+
+BOOST_AUTO_TEST_CASE(creating_an_STS_with_unsup_having_a_wrong_size_leads_to_exception)
+{
+    std::vector<double> unsupE(5, 0.);
+    STS_setup<4> sts_setup;
+    std::string err_msg = "Remix hydro input : arrays of different sizes";
+    BOOST_CHECK_EXCEPTION(sts_setup.createSTS(unsupE),
+                          std::invalid_argument,
+                          checkMessage(err_msg));
+}
+
+BOOST_AUTO_TEST_CASE(creating_2_STS_of_different_sizes___checking_input_of_algo_leads_to_exception)
+{
+    std::vector<double> Load(5);
+    std::vector<double> TotaGenWithoutStorage(5);
+    std::vector<double> Spillage(5);
+    std::vector<double> DTG_MRG(5);
+
+    // Creating sts_1 of size 5
+    std::vector<double> unsupE_1(5, 0.);
+    STS_setup<5> sts_setup_1;
+    auto sts_1 = sts_setup_1.createSTS(unsupE_1);
```

Comment:
`STS_setup` does nothing but hold data


```suggestion
    auto sts_1 = createSTS(unsupE_1);
```

Reply:
I renamed **STS_setup** into **STS_holder**, **STS_setup** being inaccurate indeed